db/hybrid_logical_vector.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package db
+
+type HybridLogicalVector struct {
+	CurrentVersionCAS uint64            // current version cas (or cvCAS) stores the current CAS at the time of replication
+	SourceID          string            // source bucket uuid of where this entry originated from
+	Version           uint64            // current cas of the current version on the version vector
+	MergeVersions     map[string]uint64 // map of merge versions for fast efficient lookup
+	PreviousVersions  map[string]uint64 // map of previous versions for fast efficient lookup
+}
+
+// CurrentVersionVector is a structure used to add a new sourceID:CAS entry to a HLV
+type CurrentVersionVector struct {
+	VersionCAS uint64
+	SourceID   string
+}
+
+// GetCurrentVersion return the current version vector from the HLV in memory
+func (hlv *HybridLogicalVector) GetCurrentVersion() (string, uint64) {
+	return hlv.SourceID, hlv.Version
+}
+
+// IsInConflict tests to see if in memory HLV is conflicting with another HLV
```

Comment:
HLV A is in conflict with HLV B if:
- A's current version dominates B's version for A's current source, OR
- B's current version dominates A's version for B's current source 

I think this is a simpler evaluation than what you're currently doing in isDominating (although there's some additional work to look up the source version across cv/pv/mv).

Reply:
I agree making those changes now! Thanks for clearing that up!